### PR TITLE
Support for Confirmed Incident Lineups

### LINKED ISSUE #12: Actual grunt data
Example grunt - showing line up and reward

```
OpenInvasionCombatSessionProto          incident_lookup { incident_id: "5477255916644222015" fort_id: "d36fb83acc5f4516bcf2e3c083145f65.16" fort_lat: 54.090941 fort_lng: 13.397025 } step: 1 attacking_pokemon_id: 6253405486787881069 attacking_pokemon_id: 7051996254806144273 attacking_pokemon_id: 4049703279889415729 lobby_join_time_ms: 1659221762713
    Response | OpenInvasionCombatSessionOutProto      status: SUCCESS combat { combat_state: ACTIVE combat_id: "COMBAT_1659221782019_-5139414284709935119" player { public_profile { name: "Flux4218" level: 30 avatar { avatar: 1 avatar_hair: "AVATAR_f_hair_default_0" avatar_shirt: "AVATAR_f_shirt_default_5" avatar_pants: "AVATAR_f_pants_default_4" avatar_hat: "AVATAR_f_hat_default_A_5" avatar_shoes: "AVATAR_f_shoes_default_1" avatar_eyes: "AVATAR_f_eyes_0" avatar_backpack: "AVATAR_f_backpack_default_0" avatar_gloves: "AVATAR_f_gloves_default_0" avatar_socks: "AVATAR_f_socks_default_0" avatar_belt: "AVATAR_f_belt_default_4" avatar_necklace: "AVATAR_f_necklace_default_0" } caught_pokemon: 20 experience: 2104205 } active_pokemon { pokemon_id: 6253405486787881069 pokedex_id: ANORITH cp: 1094 cp_multiplier: 0.7068842 stamina: 96 max_stamina: 96 move1: SCRATCH_FAST move2: ANCIENT_POWER pokemon_display { gender: FEMALE form: ANORITH_NORMAL display_id: -2902952500726692630 } individual_attack: 10 individual_defense: 2 individual_stamina: 8 pokeball: ITEM_ULTRA_BALL } reserve_pokemon { pokemon_id: 7051996254806144273 pokedex_id: AIPOM cp: 995 cp_multiplier: 0.7068842 stamina: 108 max_stamina: 108 move1: SCRATCH_FAST move2: AERIAL_ACE pokemon_display { gender: FEMALE form: AIPOM_NORMAL display_id: 2480148140930035634 } individual_attack: 15 individual_defense: 1 individual_stamina: 8 pokeball: ITEM_ULTRA_BALL } reserve_pokemon { pokemon_id: 4049703279889415729 pokedex_id: RAMPARDOS cp: 2054 cp_multiplier: 0.65443563 stamina: 146 max_stamina: 146 move1: ZEN_HEADBUTT_FAST move2: FLAMETHROWER pokemon_display { gender: MALE display_id: 4961293038146386647 } individual_attack: 12 individual_stamina: 5 pokeball: ITEM_ULTRA_BALL } minigame_defense_chances_left: 2 lobby_join_time_ms: 1659221762713 } opponent { public_profile { name: "CHARACTER_FIGHTING_GRUNT_FEMALE" } active_pokemon { pokedex_id: MAKUHITA cp: 1754 cp_multiplier: 1.0204883 stamina: 116 max_stamina: 116 move1: ROCK_SMASH_FAST move2: CROSS_CHOP pokemon_display { gender: MALE alignment: SHADOW } individual_attack: 91 individual_defense: 15 individual_stamina: 9 } reserve_pokemon { pokedex_id: HITMONCHAN cp: 5004 cp_multiplier: 1.0204883 stamina: 92 max_stamina: 92 move1: BULLET_PUNCH_FAST move2: CLOSE_COMBAT pokemon_display { gender: MALE alignment: SHADOW } individual_attack: 153 individual_defense: 15 individual_stamina: 9 } reserve_pokemon { pokedex_id: MACHOKE cp: 4373 cp_multiplier: 1.0204883 stamina: 125 max_stamina: 125 move1: LOW_KICK_FAST move2: BRICK_BREAK pokemon_display { gender: MALE alignment: SHADOW } individual_attack: 143 individual_defense: 15 individual_stamina: 9 } minigame_defense_chances_left: 2 combat_npc_personality_id: "TRAINER_PERSONALITY_EASY" } combat_start_ms: 1659221792019 combat_end_ms: 1659222062019 server_ms: 1659221782019 }
```

### LINKED ISSUE #80: Invasion Teams & Giovanni detection
I would like to discuss how to process invasion teams & confirmed Giovanni

Refer to #12 for real invasion data examples.

# Giovanni

## Raw Data

This data is best extracted from a `StartIncidentOutProto`.

`StartIncidentOutProto.Incident.step` is an array of `ClientIncidentStepProto`. In it, the confirmed grunt character is in both `pokestop_dialogue.dialogue_line.character` (`ClientPokestopNpcDialogueStepProto`) and `invasion_battle.character` (`ClientInvasionBattleStepProto`)

A confirmed Giovanni is type `44`. Male Decoy is `45` and Female Decoy is `46` (used by the game). Note that "unconfirmed" Giovanni is also `44`

## Processing

- Add a new bool column to the `incident` table. I suggest calling it `character_confirmed`
- After processing this, send a new webhook. This can then also have the same `character_confirmed` bool field.

## Notes

- There's a relatively fresh `_u` flag for grunt UICONs. This is part of the standard but afaik, no uicon repo supports it. Reference PR: Uicons/UIcons#34

# Teams

## Raw Data

This data is provided in a `OpenInvasionCombatSessionOutProto`.

`OpenInvasionCombatSessionOutProto.combat.opponent.active_pokemon` is the first slot, OpenInvasionCombatSessionOutProto.combat.opponent.reserve_pokemon is an array containing the second and third slot (in order).

These Pokemon are of type `CombatPokemonProto`. Refer to #12 for details on this. 

You may want to abort processing if the array size != 2. This should never happen. Also only process OpenInvasionCombatSessionOutProtos that have teh SUCCESS status. It's not uncommon these can fail.

## Processing

I'm not sure how to best tackle this.

### Pokemon Data

- Useful: Pokedex ID and Form Id
- Possibly useful: Gender (not sure if same across players)
- Possibly useful: Moves (not sure if same across players), possibly interesting for hardcore players, but most likely not worth storing
- Likely useless: CP/CPM/IVs. Most likely different across players

### Storing

- If we only decide to store pokedexId and formId, we would get away with adding 6 new columns to the `incidents` table: `slot_x_pokemon_id` and `slot_x_form` where x is 0, 1, 2 (or 1, 2, 3)
- If we decide to store more, we would probably have to make a new table. This will become quite unperformant and kind of ugly
- When a team is detected, send out a incident webhook with a field that should look like this:

```js
team: [
    {
        "pokemon_id": 0,
        "form": 0,
        ...
    },
    ...
]
```

### PR DESCRIPTION
**Pull request by TurtIeSocks**
Golbat support to decode proto methods 1200 and 1202

**1200** can confirm the incident character, such as Giovanni or a Decoy
**1202** confirms the entire incident's lineup to inform users about the reward

Webhook Additions:
```json
    {
      "confirmed": true,
      "lineup": [
        { "slot": 1, "pokemon_id": 2, "form": 3 },
        { "slot": 2, "pokemon_id": 124, "form": 124 },
        { "slot": 3, "pokemon_id": 97, "form": 42 }
      ]
    }
```
Screenshot 2023-04-04 at 11 25 54 PM

Change Notes:
- migration to add some columns to the incident table
- update invasion lineups and confirmation status when they're opened
- sends webhooks
- requires mitm that sends these protos
- 1202 also requires the mitm to send `OpenInvasionCombatSessionProto` (the request) in order to confirm the `incident_id` for the decoder

Closes #12
Closes #80 

Confirmed working with https://github.com/WatWowMap/ReactMap/pull/682

### CODE REVIEW

File: decoder/incident.go [hunk 1]
```diff
@@ -66,7 +80,15 @@ func hasChangesIncident(old *Incident, new *Incident) bool {
 		old.DisplayType != new.DisplayType ||
 		old.Style != new.Style ||
 		old.Character != new.Character ||
-		old.Updated != new.Updated
+		old.Confirmed != new.Confirmed ||
```
Comment (na-ji) [inline decoder/incident.go:83]: We should probably prevent any change if `Confirmed` is true. This way, we prevent a detected Giovanni/Decoy to be reverted to a regular Giovanni.

Comment (jfberry) [reply decoder/incident.go:83]: The fix should be in updateFromPokestopIncidentDisplay -- to not actually make the change from the GMO if it is previously confirmed (or whatever the exact rule should be)

File: decoder/incident.go [hunk 2]
```diff
@@ -118,7 +147,22 @@ func createIncidentWebhooks(ctx context.Context, db db.DbDetails, oldIncident *I
 		if stop == nil {
 			stop = &Pokestop{}
 		}
-
+		lineup := [3]webhookLineup{}
```
Comment (jfberry) [inline decoder/incident.go:150]: Simpler syntax for this - see playground https://go.dev/play/p/NFIwWJ4rD8L

File: decoder/incident.go (hunk 1 above)
Comment (TurtIeSocks) [reply decoder/incident.go:83]: Resolved in latest.

File: decoder/incident.go (hunk 2 above)
Comment (TurtIeSocks) [reply decoder/incident.go]: Resolved in latest, thanks for the playground!

**Comment by jfberry** [comment]
Lgtm

**Pull request merged**